src/pages/ScholarshipForm.jsx

```diff
@@ -0,0 +1,202 @@
+        className={classes.fieldStyle}
+        type="number"
+        InputLabelProps={{ shrink: true }}
+        fullWidth
+        InputProps={{ inputProps: { min: 0, step: 50 } }}
+        value={formFieldStates[field]}
+        onChange={(e) =>
+          handleFieldChange(field, parseInt(e.target.value, 10) || '')
+        }
+      />
+    ));
+  }
+
+  function displayField(fieldKey) {
+    const { key } = ScholarshipFields[fieldKey];
+    const { label } = ScholarshipFields[fieldKey];
+    const { type } = ScholarshipFields[fieldKey];
+
+    switch (type) {
+      case 'text':
+      case 'number':
+        return (
+          <TextField
+            key={key}
+            id={fieldKey}
+            label={label}
+            className={classes.fieldStyle}
+            type={type === 'text' ? 'text' : 'number'}
+            InputLabelProps={{ shrink: true }}
+            fullWidth
+            InputProps={{ inputProps: { min: 0, step: 50 } }}
+            value={formFieldStates[fieldKey]}
+            onChange={(e) =>
+              handleFieldChange(
+                fieldKey,
+                type === 'number'
+                  ? parseInt(e.target.value, 10)
+                  : e.target.value
+              )
+            }
+          />
+        );
+
+      case 'date':
+        return (
+          <MuiPickersUtilsProvider utils={DateFnsUtils} key={key}>
+            <KeyboardDatePicker
+              id={fieldKey}
+              label={label}
+              value={formFieldStates[fieldKey]}
+              onChange={(date) => handleFieldChange(fieldKey, date)}
+              format="MM/dd/yyyy"
+            />
+          </MuiPickersUtilsProvider>
+        );
+
+      case 'select':
+        return (
+          <FormControl key={key} className={classes.formControlStyle}>
+            <InputLabel shrink>{label}</InputLabel>
+            <Select
+              label={label}
+              id={fieldKey}
+              value={formFieldStates[fieldKey]}
+              onChange={(e) => handleFieldChange(fieldKey, e.target.value)}>
+              {ScholarshipFields[fieldKey].options.map((option) => {
+                return (
+                  <MenuItem key={option} value={option}>
+                    {option}
+                  </MenuItem>
+                );
+              })}
+            </Select>
+          </FormControl>
+        );
+
+      case 'checkbox':
+        return (
+          <FormControl key={key}>
+            <FormControlLabel
+              value={ScholarshipFields[fieldKey]}
+              onChange={(e) => handleFieldChange(fieldKey, e.target.checked)}
+              control={<Checkbox color="primary" />}
+              label={label}
+              labelPlacement="start"
+            />
+          </FormControl>
+        );
+
+      case 'amountSub':
+        return displayAmountFields();
+
+      default:
+        return <div />;
+    }
+  }
+
+  return (
+    <Container style={{ width: '40%' }}>
```

Comment:
This may not look great on mobile devices. I suggest maybe leaving this as is or finding a different way to set width for just desktop

Reply:
This wasn't updated